tests/accel/upgrade/tests/upgrade.go

```diff
@@ -0,0 +1,177 @@
+package upgrade
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/golang/glog"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"github.com/openshift-kni/eco-goinfra/pkg/clusteroperator"
+	"github.com/openshift-kni/eco-goinfra/pkg/clusterversion"
+	"github.com/openshift-kni/eco-goinfra/pkg/namespace"
+	"github.com/openshift-kni/eco-goinfra/pkg/pod"
+
+	"github.com/openshift-kni/eco-goinfra/pkg/reportxml"
+	"github.com/openshift-kni/eco-goinfra/pkg/route"
+	"github.com/openshift-kni/eco-gotests/tests/accel/internal/accelconfig"
+	accelinittools "github.com/openshift-kni/eco-gotests/tests/accel/internal/accelinittools"
+	"github.com/openshift-kni/eco-gotests/tests/accel/upgrade/internal/accelparams"
+
+	"github.com/openshift-kni/eco-gotests/tests/internal/url"
+)
+
+var (
+	waitToUpgradeStart     = 5 * time.Minute
+	waitToUpgradeCompleted = 130 * time.Minute
+	desiredUpgradeChannel  = "stable-" + accelinittools.AccelConfig.HubMinorVersion
+	stream                 = accelparams.Z
+)
+
+var _ = Describe("OCP_UPGRADE", Ordered, Label("minor"), func() {
+	Context("OCP", func() {
+
+		AfterAll(func() {
+			By("Delete workload test namespace")
+			glog.V(90).Infof("Deleting test deployments")
+			deleteWorkloadNamespace()
+		})
+
+		It("should upgrade successfully", reportxml.ID("72245"), func() {
+			By("Get the clusterversion struct")
+			version, err := clusterversion.Pull(accelinittools.HubAPIClient)
+			Expect(err).ToNot(HaveOccurred(), "error retrieving clusterversion")
+			glog.V(90).Infof("got the clusterversion struct %+v", version)
+
+			By("Deploy a workload in the cluster, expose a service and create a route")
+			workloadRoute := startTestWorkloadAndGetRoute()
+
+			By("Patch the clusterversion with the desired upgrade channel")
+			if desiredUpgradeChannel == "stable-" {
+				desiredUpgradeChannel = version.Object.Spec.Channel
+				glog.V(90).Infof("clusterversion channel %s", desiredUpgradeChannel)
+			}
+			version, err = version.WithDesiredUpdateChannel(desiredUpgradeChannel).Update()
+			Expect(err).ToNot(HaveOccurred(), "error patching the desired upgrade channel")
+			glog.V(90).Infof("patched the clusterversion channel %s", desiredUpgradeChannel)
+
+			By("Get the desired update image")
+			config := accelconfig.NewAccelConfig()
```

Comment:
You don't need to initial NewAccelConfig it's already available in accelinittools.